fastly/objectstorage/accesskeys/api_test.go

```diff
@@ -0,0 +1,161 @@
+		accessKeys, err = ListAccessKeys(c)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Make sure the test AccessKey we're going to create isn't among them.
+	for _, ak := range accessKeys.Data {
+		if ak.Description == TestAccessKeyDescription &&
+			ak.Permission == TestAccessKeyPermission &&
+			len(ak.Buckets) == len(TestAccessKeyBuckets) &&
+			ak.Buckets[0] == TestAccessKeyBuckets[0] {
+			t.Errorf("found test AccessKey %q, aborting", ak.AccessKeyID)
+		}
+	}
+
+	// Create a AccessKey for testing.
+	var accessKey *AccessKey
+	fastly.Record(t, "create_AccessKey", func(c *fastly.Client) {
+		accessKey, err = Create(c, &CreateInput{
+			Description: fastly.ToPointer(TestAccessKeyDescription),
+			Permission:  fastly.ToPointer(TestAccessKeyPermission),
+			Buckets:     fastly.ToPointer(TestAccessKeyBuckets),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if accessKey.Description != TestAccessKeyDescription {
+		t.Errorf("unexpected AccessKey name: got %q, expected %q", accessKey.Description, TestAccessKeyDescription)
+	}
+	if accessKey.Permission != TestAccessKeyPermission {
+		t.Errorf("unexpected AccessKey permission: got %q, expected %q", accessKey.Permission, TestAccessKeyPermission)
+	}
+	if len(accessKey.Buckets) != len(TestAccessKeyBuckets) {
+		t.Errorf("unexpected AccessKey buckets length: got %q, expected %q", len(accessKey.Buckets), len(TestAccessKeyBuckets))
+	}
+	if accessKey.Buckets[0] != TestAccessKeyBuckets[0] {
+		t.Errorf("unexpected AccessKey bucket: got %q, expected %q", accessKey.Buckets[0], TestAccessKeyBuckets[0])
+	}
+
+	// Ensure we delete the test AccessKey at the end.
+	defer func() {
+		fastly.Record(t, "delete_AccessKey", func(c *fastly.Client) {
+			err = Delete(c, &DeleteInput{
+				AccessKeyID: fastly.ToPointer(accessKey.AccessKeyID),
+			})
+		})
+		if err != nil {
+			t.Errorf("error during AccessKey cleanup: %v", err)
+		}
+	}()
+
+	// Get the test AccessKey.
+	var ac *AccessKey
+	fastly.Record(t, "get_AccessKey", func(c *fastly.Client) {
+		ac, err = Get(c, &GetInput{
+			AccessKeyID: fastly.ToPointer(accessKey.AccessKeyID),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if ac.Description != accessKey.Description {
+		t.Errorf("unexpected AccessKey Description: got %q, expected %q", ac.Description, accessKey.Description)
+	}
+	if ac.Permission != accessKey.Permission {
+		t.Errorf("unexpected AccessKey Permissions: got %q, expected %q", ac.Permission, accessKey.Permission)
+	}
+	if len(ac.Buckets) != len(accessKey.Buckets) {
+		t.Errorf("unexpected AccessKey Buckets length: got %q, expected %q", len(ac.Buckets), len(accessKey.Buckets))
+	}
+	if ac.Buckets[0] != accessKey.Buckets[0] {
+		t.Errorf("unexpected AccessKey Buckets contents: got %q, expected %q", ac.Buckets[0], accessKey.Buckets[0])
+	}
+
+	// List all entries of the test AccessKey and compare it to the input.
+	var actualAccessKeys *AccessKeys
+	fastly.Record(t, "list_accesskeys_with_new", func(c *fastly.Client) {
```

Comment:
I would recommend to remove `accesskeys`/`Accesskey` from the recording names and just keep it simple with `create`, `delete`, `get`, `list`, and `list_with_new_value`.